linkis-public-enhancements/linkis-script-dev/linkis-storage-script-dev-server/src/main/java/org/apache/linkis/filesystem/restful/api/FsRestfulApi.java

```diff
@@ -583,12 +587,33 @@ public Message openFile(
       if (FileSource.isResultSet(fsPath.getPath())) {
         fileSource = fileSource.page(page, pageSize);
       }
+      if (limitTotalLine > 0) {
+        fileSource =
+            fileSource.limitTotalLine(
+                Math.min(limitTotalLine, FILESYSTEM_LIMIT_TOTAL_LINE.getValue()));
+      }
+      if (limitBytes > 0) {
+        fileSource = fileSource.limitBytes(Math.min(limitBytes, FILESYSTEM_LIMIT_BYTES.getValue()));
+      }
+      if (limitColumnLength > 0) {
+        fileSource =
+            fileSource.limitColumnLength(
+                Math.min(limitColumnLength, FILESYSTEM_LIMIT_COLUMN_LENGTH.getValue()));
+      }
       Pair<Object, List<String[]>> result = fileSource.collect()[0];
       IOUtils.closeQuietly(fileSource);
-      message.data("metadata", result.getFirst()).data("fileContent", result.getSecond());
+      message.data("metadata", result.getFirst());
+      message.data("fileContent", result.getSecond());
       message.data("type", fileSource.getFileSplits()[0].getType());
       message.data("totalLine", fileSource.getTotalLine());
-      return message.data("page", page).data("totalPage", 0);
+      message.data("totalCount", fileSource.getTotalCount());
```

Comment:
It is not recommended to return the total number of lines directly, it is recommended to call fileinfo：org.apache.linkis.filesystem.restful.api.FsRestfulApi#fileInfo

Reply:
Ok, good advice